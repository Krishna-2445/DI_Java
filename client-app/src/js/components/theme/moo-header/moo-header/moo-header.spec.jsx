import React from 'react';
import { shallow } from 'enzyme';

import MooHeader from './moo-header';

describe('<MooHeader />', () => {
  it('renders as expected with siteTitle prop', () => {
    const wrapper = shallow(<MooHeader siteTitle="Site Title" />);
    expect(wrapper.contains('Site Title')).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders children as expected', () => {
    const child = <div>Child</div>;
    const wrapper = shallow(<MooHeader siteTitle="Site Title"> {child} </MooHeader>);
    expect(wrapper.contains('Child')).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with no props passed in', () => {
    const wrapper = shallow(<MooHeader />);
    expect(wrapper.exists()).toEqual(true);
  });

  it('displays horizontal logo when there is no siteTitle', () => {
    const wrapper = shallow(<MooHeader />);
    expect(wrapper.find('.horizontal-logo').exists()).toEqual(true);
  });
});
