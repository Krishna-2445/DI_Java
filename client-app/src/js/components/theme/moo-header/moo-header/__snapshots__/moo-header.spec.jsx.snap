// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MooHeader /> renders as expected with siteTitle prop 1`] = `
<div
  className="moo-header-wrapper"
>
  <div
    className="moo-header"
  >
    <NavLink
      activeClassName="active"
      to="/"
    >
      <img
        className="logo"
        src="test-file-stub"
      />
      <h1>
        Site Title
      </h1>
    </NavLink>
  </div>
</div>
`;

exports[`<MooHeader /> renders children as expected 1`] = `
<div
  className="moo-header-wrapper"
>
  <div
    className="moo-header"
  >
    <NavLink
      activeClassName="active"
      to="/"
    >
      <img
        className="logo"
        src="test-file-stub"
      />
      <h1>
        Site Title
      </h1>
    </NavLink>
     
    <div>
      Child
    </div>
     
  </div>
</div>
`;
